src/main/resources/avro/common.avdl

```diff
@@ -5,44 +5,45 @@ not have any methods; it is merely a library of types.
 
 ##Sequences, Segments, and References
 
-A sequence is a component in a sequence graph, optionally joined at its ends to
-other sequences. Sequences are not directly retrievable through the API, but
-ranges of their bases can be downloaded.
-
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. When you ask about the novel
-sequences that come in a `VariantSet`, you will get back `Segment`s covering
-them. A `Path` through a sequence graph is a list of `Segment`s.
-
-A `Reference` (defined in references.avdl) is an object that marks a sequence as
-part of a `ReferenceSet`'s sequence graph. It contains a `Segment` covering the
-sequence it belongs to.
+A `Sequence` is a component in a sequence graph, optionally joined to other
+`Sequence`s via `Join`s. It has to have length at least 1, but does not contain
+base data itself - the base data is obtained with the `getSequenceBases()`
+method call.
+
+A `Segment` is a range on a `Sequence`. When you ask about the novel
+`Sequence`s that come in a `VariantSet`, you will get back `Segment`s covering
+them. A `Path` through a sequence graph is a list of `Segment`s, with the
+requirement that adjacent segments are joined by `Join`s.
+
+A `Reference` (defined in references.avdl) is part of a `ReferenceSet` which
+represents the reference sequence graph. It contains a `Segment` identifying
+the underlying `Sequence`. The reason we do this rather than directly containing
+the `Sequence` is to allow generation of local `Reference`s and `ReferenceSet`s
+from part of a larger one, e.g. the MHC region of chromosome 6.
 
 ##Sequence Graph Concepts
 
 Many of the types defined here are related to defining sequence graphs. Sequence
 graph concepts can be illustrated on a piece of double-stranded DNA with two
 nicks in it, one on the top strand and one on the bottom strand. Here is an
-example where the top strand is GGTGGNG.
-
-```
-            | <- top strand nick at position (2,+)
-5' -------  -------------------  3'
-     G   G   T   G   G   N   G
-     C   C   A   C   C   N   C
-3' ----------------  ----------  5'
-                  | <- bottom strand nick at position (3,-)
-
-    +0- +1- +2- +3- +4- +5- +6- <- coordinates
-```
-
-A sequence is a piece of double-stranded DNA composed of a series of DNA
-basepairs. In the default forward orientation a sequence is specified by the DNA
-letters of its top strand, e.g. in this case GGTGGNG, where N indicates an
+example where the top strand is GGTGGAG.
+
```

Comment:
Sorry, when I was editing I half edited N to A, then stopped.  I think it would be good to remove N from
the examples, although to say somewhere that N is a legitimate character in a sequence.  There are a few
Ns, but they are not typical.

Richard

> On 17 Mar 2015, at 04:40, Maciek Smuga-Otto <notifications@github.com> wrote:
> 
> In src/main/resources/avro/common.avdl <https://github.com/ga4gh/schemas/pull/250#discussion_r26549104>:
> 
> > -```
> > -            | <- top strand nick at position (2,+)
> > -5' -------  -------------------  3'
> > -     G   G   T   G   G   N   G
> > -     C   C   A   C   C   N   C
> > -3' ----------------  ----------  5'
> > -                  | <- bottom strand nick at position (3,-)
> > -
> > -    +0- +1- +2- +3- +4- +5- +6- <- coordinates
> > -```
> > -
> > -A sequence is a piece of double-stranded DNA composed of a series of DNA
> > -basepairs. In the default forward orientation a sequence is specified by the DNA
> > -letters of its top strand, e.g. in this case GGTGGNG, where N indicates an
> > +example where the top strand is GGTGGAG.
> > +
> Example below still uses GGTGGNG. Do you want to change N->A throughout the rest of the example?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ga4gh/schemas/pull/250/files#r26549104>.
> 




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.